src/components/ChannelPreview/ChannelPreview.js

```diff
@@ -1,107 +1,60 @@
-    channel.off('message.new', this.handleNewMessageEvent);
-    channel.off('message.read', this.handleReadEvent);
-  }
-
-  handleReadEvent = (event) => {
-    if (event.user.id === this.props.client.userID) {
-      this.setState({ unread: this.props.channel.countUnread() });
-    }
-  };
+  const [unread, setUnread] = useState(channel.countUnread());
+  const [lastMessage, setLastMessage] = useState({});
+  const latestMessagePreview = useLatestMessagePreview(channel);
 
-  handleNewMessageEvent = (event) => {
-    const channel = this.props.channel;
-    this.setState({
-      lastMessage: event.message,
-      unread: channel.countUnread(),
-    });
-  };
+  useEffect(() => {
+    const handleNewMessageEvent = (e) => {
+      setLastMessage(e.message);
+      setUnread(channel.countUnread());
+    };
 
-  getLatestMessage = () => {
-    const { channel, t, tDateTimeParser } = this.props;
-    const message = channel.state.messages[channel.state.messages.length - 1];
+    channel.on('message.new', handleNewMessageEvent);
 
-    const latestMessage = {
-      text: '',
-      created_at: '',
-      messageObject: { ...message },
+    return () => {
+      channel.off('message.new', handleNewMessageEvent);
     };
+  });
 
-    if (!message) {
-      latestMessage.text = t('Nothing yet...');
-      return latestMessage;
-    }
-    if (message.deleted_at) {
-      latestMessage.text = t('Message deleted');
-      return latestMessage;
-    }
-
-    if (message.text) {
-      latestMessage.text = message.text;
-    } else {
-      if (message.command) {
-        latestMessage.text = '/' + message.command;
-      } else if (message.attachments.length) {
-        latestMessage.text = t('🏙 Attachment...');
-      } else {
-        latestMessage.text = t('Empty message...');
+  useEffect(() => {
+    const handleReadEvent = (e) => {
+      if (e.user.id === client.userID) {
+        setUnread(0);
       }
-    }
-
-    if (tDateTimeParser(message.created_at).isSame(new Date(), 'day'))
-      latestMessage.created_at = tDateTimeParser(message.created_at).format(
-        'LT',
-      );
-    else {
-      latestMessage.created_at = tDateTimeParser(message.created_at).format(
-        'L',
-      );
-    }
-
-    return latestMessage;
-  };
+    };
 
-  render() {
-    const props = { ...this.state, ...this.props };
-    const { Preview } = this.props;
-    return <Preview {...props} latestMessage={this.getLatestMessage()} />;
-  }
-}
+    channel.on('message.read', handleReadEvent);
 
-export default withTranslationContext(ChannelPreview);
+    return () => {
+      channel.off('message.read', handleReadEvent);
+    };
+  });
+
+  const { Preview } = props;
+  return (
+    <Preview
+      {...props}
+      lastMessage={lastMessage}
+      unread={unread}
+      latestMessage={latestMessagePreview}
+    />
+  );
+};
+
+ChannelPreview.propTypes = {
+  channel: PropTypes.object.isRequired,
+  client: PropTypes.object.isRequired,
```

Comment:
this is no longer a required prop as we're pulling it from ChatContext, right?

Reply:
true, and we're going to axe all these prop types anyways shortly with TS